build/android/bytecode/java/org/brave/bytecode/BraveClassAdapter.java

```diff
@@ -0,0 +1,15 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package org.brave.bytecode;
+
+import org.objectweb.asm.ClassVisitor;
+
+public class BraveClassAdapter {
+    public static ClassVisitor createAdapter(ClassVisitor chain) {
+        chain = new BraveBookmarkModelClassAdapter(chain);
+        chain = new BraveMainPreferenceBaseClassAdapter(chain);
```

Comment:
do we override the chain? I think the replacing happens for the latest only

Reply:
no, it returns the same chain with the new class added to it